contracts/FragToken.sol

```diff
@@ -9,15 +9,23 @@ import "openzeppelin-solidity/contracts/access/Ownable.sol";
 
 contract FRAGToken is ERC20, ERC20Permit, Ownable {
     uint8 constant DECIMALS = 12; // Preferred for Fragnova (Substrate)
-    uint256 constant INITIAL_SUPPLY = 10_000_000_000 * (10**DECIMALS);
+    uint256 constant INITIAL_SUPPLY = 10_000_000_000 * (10**DECIMALS); 
+    uint256 private constant _TIMELOCK = 1 weeks;
 
     mapping(address => uint256) private _locksAmount;
     mapping(address => uint256) private _locksBlock;
-
-    uint256 private _lockCooldown = 45500; // Roughly 1 week
+    mapping(address => uint256) private _locktime;
+
+    enum Period {
+        TwoWeeks,
+        OneMonth,
+        ThreeMonths,
+        SixMonths,
+        OneYear
+    }
 
     // Fragnova chain will listen to those events
-    event Lock(address indexed sender, bytes signature, uint256 amount);
+    event Lock(address indexed sender, bytes signature, uint256 amount, uint256 timelock);
```

Comment:
Now that I see this, we need to make sure that clamor side this `timelock` is ignored when `Unlock` happens, not sure if that's the case. Also event HASH has changed (I think it's a constant clamor side)

Reply:
Correct, the event hash has changed too. Need to change on Clamor side.